From 2bfabfa5aaa458a7bb96833de1bc5d2b4db9aa7e Mon Sep 17 00:00:00 2001
From: feeRnt <81442162+feeRnt@users.noreply.github.com>
Date: Sat, 30 Aug 2025 16:15:02 +0000
Subject: [PATCH] drm: amdgpu/ps4_bridge - Use MODE_TYPE_PREFERRED

Try using MODE_TYPE_PREFERRED in ps4_bridge.c,
only for the 1920x1080-60 Hz mode.

This may allow using higher display modes and refresh
rates (60Hz vs 120Hz) without compromising on support
for older monitors and lower resolutions.
---
 drivers/gpu/drm/amd/amdgpu/ps4_bridge.c | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/ps4_bridge.c b/drivers/gpu/drm/amd/amdgpu/ps4_bridge.c
index e1f7807749831ee63291b21f734731bf589b4daf..117c8322d8449ae72900776c5318d2fa33e3ae12 100644
--- a/drivers/gpu/drm/amd/amdgpu/ps4_bridge.c
+++ b/drivers/gpu/drm/amd/amdgpu/ps4_bridge.c
@@ -688,7 +688,7 @@ static const struct drm_display_mode mode_720p = {
 };
 /* 16 - 1920x1080@60Hz */
 static const struct drm_display_mode mode_1080p = {
-	DRM_MODE("1920x1080", DRM_MODE_TYPE_DRIVER, 148500, 1920, 2008,
+	DRM_MODE("1920x1080", DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED, 148500, 1920, 2008,
 		 2052, 2200, 0, 1080, 1084, 1089, 1125, 0,
 		 DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC),
 	.picture_aspect_ratio = HDMI_PICTURE_ASPECT_16_9
@@ -698,17 +698,15 @@ static const struct drm_display_mode mode_1080p = {
  * with a 120Hz mode - leading to a blackscreen.
  * Only fix seems to be having a xorg.conf in /usr/share/X11/xorg.conf.d/
  *
- * Disable for now
+ * Try setting a TYPE_PREFFERED mode
  */
 /* 63 - 1920x1080@120Hz */
-/*
 static const struct drm_display_mode mode_1080p120 = {
 	DRM_MODE("1920x1080", DRM_MODE_TYPE_DRIVER, 297000, 1920, 2008,
 			2052, 2200, 0, 1080, 1084, 1089, 1125, 0,
 		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC),
 	  .picture_aspect_ratio = HDMI_PICTURE_ASPECT_16_9
 };
-*/
 
 int ps4_bridge_get_modes(struct drm_connector *connector)
 {
@@ -719,10 +717,8 @@ int ps4_bridge_get_modes(struct drm_connector *connector)
 	newmode = drm_mode_duplicate(dev, &mode_1080p);
 	drm_mode_probed_add(connector, newmode);
 
-	/*
 	newmode = drm_mode_duplicate(dev, &mode_1080p120);
 	drm_mode_probed_add(connector, newmode);
-	*/
 
 	//newmode = drm_mode_duplicate(dev, &mode_720p);
 	//drm_mode_probed_add(connector, newmode);
@@ -771,13 +767,14 @@ enum drm_mode_status ps4_bridge_mode_valid(struct drm_connector *connector,
 	int vic = drm_match_cea_mode(mode);
 
 	/* Allow anything that we can match up to a VIC (CEA modes) */
-	//if (!vic || (vic != 16 && vic != 4 && vic != 63)) {
-	//Disabled 1920x1080-120Hz for now
+	if (!vic || (vic != 16 && vic != 4 && vic != 63)) {
+	// Might need to disable 63 (1920x1080-120Hz)
 
+	/*
 	if (!vic || (vic != 16 && vic != 4)) {
+	*/
 		return MODE_BAD;
 	}
-
 	return MODE_OK;
 }
 
